File: app/src/main/java/my/ew/wallpaper/utils/PreferencesHelper.java
package my.ew.wallpaper.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.util.Log;

/**
 * Created by isanechek on 24.07.15.
 */
public class PreferencesHelper {

    private static final String TAG = "pref_helper";

    private static final String SETTINGS = "wallpaper_prefs_a";
    public static final String PREF_WELCOME_DONE = "pref_welcome_done";
    private static final String GAPPS_SETTING = "gappspref";
    private static final String TAG_DISABLED_ADS = "disabledADS";
    protected static final String WALLPAPER_WIDTH_KEY = "wallpaper.width";
    protected static final String WALLPAPER_HEIGHT_KEY = "wallpaper.height";

    private static boolean disabledADS;

    public static boolean isAdsDisabled() {
        Log.d(TAG, "boolean isAdsDisabled");
//		LOG.d("disabledADS = "+disabledADS);
        return disabledADS;
    }

    public static enum Purchase {
        DISABLE_ADS
    };

    public static void savePurchase(Context c, Purchase p, boolean v) {
        Log.d(TAG, "savePurchase");
        SharedPreferences settings = c.getSharedPreferences(SETTINGS, 0);
        SharedPreferences.Editor editor = settings.edit();
        switch (p) {
            case DISABLE_ADS:
                editor.putBoolean("TAG_DISABLED_ADS", v);
                disabledADS = v;
                Log.d(TAG, "disable ads " + v);
                break;
        }
        editor.apply();
    }

    public static void loadSettings(Context c) {
        Log.d(TAG, "loadSettings");

        SharedPreferences settings = c.getSharedPreferences(SETTINGS, 0);
        if (settings.getAll().size() == 0) {
            SharedPreferences.Editor editor = settings.edit();
            editor.clear();
            editor.apply();

            disabledADS = false;
            Log.d(TAG, "load setting disable false");
        } else {
            disabledADS = settings.getBoolean(TAG_DISABLED_ADS, false);
            Log.d(TAG, "else get boolean false");
        }

    }

    public static void saveSettings(Context c) {
        Log.d(TAG, "saveSettings");
        SharedPreferences settings = c.getSharedPreferences(SETTINGS, 0);
        SharedPreferences.Editor editor = settings.edit();
        editor.putBoolean(TAG_DISABLED_ADS, disabledADS);

        editor.apply();
    }
    public static void enableGAPPS(Context c) {
        SharedPreferences settings = c.getSharedPreferences(GAPPS_SETTING, 0);
        SharedPreferences.Editor editor = settings.edit();
        editor.putBoolean("gapps", true);
        editor.apply();
    }

    public static void disableGAPPS(Context c) {
        SharedPreferences settings = c.getSharedPreferences(GAPPS_SETTING, 0);
        SharedPreferences.Editor editor = settings.edit();
        editor.putBoolean("gapps", false);
        editor.apply();
    }

    public static boolean isWelcomeDone(final Context context) {
        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(context);
        return sp.getBoolean(PREF_WELCOME_DONE, false);
    }

    public static void markWelcomeDone(final Context context) {
        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(context);
        sp.edit().putBoolean(PREF_WELCOME_DONE, true).commit();
    }

}

File: app/src/main/java/my/ew/wallpaper/utils/PreferencesHelper.kt
package my.ew.wallpaper.utils

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.util.Log

/**
 * Created by isanechek on 24.07.15.
 */
object PreferencesHelper {

    private val TAG = "pref_helper"

    private val SETTINGS = "wallpaper_prefs_a"
    public val PREF_WELCOME_DONE: String = "pref_welcome_done"
    //    private static final String GAPPS_SETTING = "gappspref";
    private val TAG_DISABLED_ADS = "disabledADS"

    private var disabledADS: Boolean = false

    @JvmStatic
    fun isAdsDisabled(): Boolean {
        Log.d(TAG, "boolean isAdsDisabled")
        return disabledADS
    }

    enum class Purchase {
        DISABLE_ADS
    }

    fun savePurchase(c: Context, p: Purchase, v: Boolean) {
        Log.d(TAG, "savePurchase")
        val settings = c.getSharedPreferences(SETTINGS, 0)
        val editor = settings.edit()
        when (p) {
            PreferencesHelper.Purchase.DISABLE_ADS -> {
                editor.putBoolean("TAG_DISABLED_ADS", v)
                disabledADS = v
                Log.d(TAG, "disable ads " + v)
            }
        }
        editor.apply()
    }

    @JvmStatic
    fun loadSettings(c: Context) {
        Log.d(TAG, "loadSettings")

        val settings = c.getSharedPreferences(SETTINGS, 0)
        if (settings.all.isNotEmpty()) {
            val editor = settings.edit()
            editor.clear()
            editor.apply()

            disabledADS = false
            Log.d(TAG, "load setting disable false")
        } else {
            disabledADS = settings.getBoolean(TAG_DISABLED_ADS, false)
            Log.d(TAG, "else get boolean false")
        }
    }

    @JvmStatic
    fun isWelcomeDone(context: Context): Boolean {
            val sp = PreferenceManager.getDefaultSharedPreferences(context)
            return sp.getBoolean(PREF_WELCOME_DONE, false)
        }

    @JvmStatic
    fun markWelcomeDone(context: Context) {
            val sp = PreferenceManager.getDefaultSharedPreferences(context)
            sp.edit().putBoolean(PREF_WELCOME_DONE, true).apply()
        }
}
